### PR TITLE
BUILD: run CI with an older windows toolchain like in wheel building

### PR DESCRIPTION
**Pull request by mattip**
issue #21346: 
- [ ] reproduce the problem with a older toolchain in CI
- [ ] fix the problem

### CODE REVIEW

**Comment by mattip** [comment]
Closing. I couldn't get the BatchScript task to actually modify the environment. I think anyway #21360 is a better fix for the issue.

File: azure-steps-windows14.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  inputs:
+    filename: tools/set_vs_toolchain.cmd
+    arguments: $(BITS) 2019 14.16
+    modifyEnvironment: True
+  displayName: Set VS toolchain version
+
+- powershell: |
+    ls openblas
+    echo What PATH and other environment variables?
+    echo PATH $Env:PATH
```
Comment (mattip) [inline azure-steps-windows14.yml:36]: For those trying to follow what happened, the PATH printed here in CI is not being modified by the previous BatchScript step, so the older 141 toolset is not used. You can see that the PATH printed at the end of `tools/set_vs_toolchain.cmd` is not the same as the one printed here.

**Comment by mattip** [comment]
@charris, @matthew-brett can we move the conversation around a CI run with vs 141 here from #21346?

I tried copying exactly the step used in MacPython/numpy-wheels, it didn't work. I printed out the PATH at the end of the step, and the PATH at the beginning of the next one, they are different. The `modifyEnvironment` directive should have made them the same.

> does the step lack a #vso[task.prependpath] set?

I don't see such a call in the code in the step in [azure/windows.yml](https://github.com/MacPython/numpy-wheels/blob/be0e819a598de3e3c637d8c9e8515048fd714273/azure/windows.yml#L23-L28) from MacPython/numpy-wheels

**Comment by h-vetinari** [comment]
> can we move the conversation around a CI run with vs 141 here from #21346?

Sidenote: it's suboptimal to do this in a closed issue/PR, because it gets hidden from much more by github, and is therefore easier to overlook. How about reopening (or moving to a different issue)?

**Pull request closed without merging**